src/common/utility.ts

```diff
@@ -292,7 +294,16 @@ export class Utility {
                     process.env[k] = envConfig[k];
                 }
             }
-        } catch (error) {}
+        } catch (error) { }
+    }
+
+    // The Azure API of listing resources is paginated. This method will follow the links and return all resources
+    public static async listAllAzureResource<T>(client: IAzureClient<T>, first: Promise<IPartialList<T>>): Promise<T[]> {
+        const all: T[] = [];
+        for (let list = await first; list.length || list.nextLink; list = list.nextLink ? await client.listNext(list.nextLink) : []) {

```

Comment:
change the condition to list !== undefined ?

Reply:
Updated. Thanks.